Pattern Library: Use a different onboarding URL on mobile

fredrikekelund opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/6534

## Proposed Changes

As reported here p1712824081983569-slack-C06ELHR6L9J, the `/setup/assembler-first` onboarding flow doesn't work on mobile. This PR makes it so `getOnboardingUrl` returns `/start` instead on mobile.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Simulate a mobile viewport
2. Navigate to `/patterns`
3. Scroll down to the `Build a site` CTA
4. Ensure that it leads to `wordpress.com/start`
5. Switch to a large viewport and reload the page
6. Ensure that the `Build a site` CTA leads to `wordpress.com/setup/assembler-first`
7. Navigate to a category
8. Scroll down to a `Get access` button and click it
9. Ensure that the Sign up and Log in links in the modal have proper query strings

fredrikekelund committed:
Use a different onboarding URL on mobile

fredrikekelund committed:
Rename variable

nightnei reviewed (approved):
LGTM and works as expected 👍
Could you please explain why do you use `URLSearchParams` instead of `buildQueryString`?

fredrikekelund commented:
> Could you please explain why do you use URLSearchParams instead of buildQueryString?

Oh yes, I meant to leave a comment about that but forgot! 🤦‍♂️

The reason I switched is that `URLSearchParams` does the same thing and it's a native API, so we save a few bytes in the build output. `PatternsGetAccessModal` was the only other place in `client/my-sites/patterns` that imported `buildQueryString`, so I changed it there as well, or else it wouldn't have made a difference in the build output size.

nightnei commented:
@fredrikekelund awesome change 👍 Personally for me it's first of all - not about "bytes", it's about - using native things as much as possible instead of "reinvented wheels".
